atomic_reactor/plugins/exit_koji_tag_build.py

```diff
@@ -0,0 +1,97 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals
+
+from atomic_reactor.constants import PLUGIN_KOJI_TAG_BUILD_KEY
+from atomic_reactor.koji_util import create_koji_session, TaskWatcher
+from atomic_reactor.plugin import ExitPlugin
+from atomic_reactor.plugins.exit_koji_promote import KojiPromotePlugin
+
+
+class KojiTagBuildPlugin(ExitPlugin):
```

Comment:
we already have plugins using other plugins,
but sure we can move it to koji_utils as long we don't have same code in 2 places

because basically as of now you can just as well call __init__ from koji promote, 
in run, have that check before create_koji_session
call login from koji_promote

and make the rest method in koji_promote and call it :)